File: web-reference/src/features/clinic-management/role-theme.ts
import type { StaffRole } from './types'

export const ROLE_ACCENTS: Record<StaffRole, string> = {
  administrator: 'from-[var(--color-violet-500)] to-[var(--color-violet-700)]',
  doctor: 'from-[var(--color-teal-500)] to-[var(--color-teal-700)]',
  receptionist: 'from-[var(--color-neutral-500)] to-[var(--color-neutral-700)]',
  lab_staff: 'from-[var(--color-neutral-400)] to-[var(--color-neutral-600)]',
}

export const ROLE_BADGE_COLORS: Record<StaffRole, 'ai' | 'teal' | 'info' | 'neutral'> = {
  administrator: 'ai',
  doctor: 'teal',
  receptionist: 'info',
  lab_staff: 'neutral',
}

/** One-line scope from settings role definitions (RBAC catalog). */
export const ROLE_SUMMARIES: Record<StaffRole, string> = {
  administrator:
    'Organization admin — staff, branches, billing, and the permission matrix.',
  doctor: 'Clinical staff — patients, appointments, visits, attachments, and AI.',
  receptionist: 'Front desk — patients, appointments, invoices, and payments.',
  lab_staff: 'Laboratory — view patients and upload visit attachments.',
}

File: frontend/lib/features/clinic-management/presentation/utils/role_theme.dart
import 'package:flutter/material.dart';

import 'package:ai_clinic/core/ui/components/app_badge.dart';
import 'package:ai_clinic/core/ui/theme/app_color_primitives.dart';
import 'package:ai_clinic/features/auth/domain/auth_session.dart';

/// Badge color per staff role (web `ROLE_BADGE_COLORS`).
const kRoleBadgeColors = <StaffRole, BadgeColor>{
  StaffRole.administrator: BadgeColor.ai,
  StaffRole.doctor: BadgeColor.teal,
  StaffRole.receptionist: BadgeColor.info,
  StaffRole.labStaff: BadgeColor.neutral,
};

@immutable
class RoleAccentGradient {
  const RoleAccentGradient({required this.start, required this.end});

  final Color start;
  final Color end;

  LinearGradient get gradient => LinearGradient(colors: [start, end]);
}

/// Column header accent gradient per role (web `ROLE_ACCENTS`).
const kRoleAccents = <StaffRole, RoleAccentGradient>{
  StaffRole.administrator: RoleAccentGradient(start: AppColorPrimitives.violet500, end: AppColorPrimitives.violet700),
  StaffRole.doctor: RoleAccentGradient(start: AppColorPrimitives.teal500, end: AppColorPrimitives.teal700),
  StaffRole.receptionist: RoleAccentGradient(start: AppColorPrimitives.neutral500, end: AppColorPrimitives.neutral600),
  StaffRole.labStaff: RoleAccentGradient(start: AppColorPrimitives.neutral400, end: AppColorPrimitives.neutral600),
};

/// One-line role scope summary for matrix header tooltips (web `ROLE_SUMMARIES`).
const kRoleSummaries = <StaffRole, String>{
  StaffRole.administrator: 'Organization admin — staff, branches, billing, and the permission matrix.',
  StaffRole.doctor: 'Clinical staff — patients, appointments, visits, attachments, and AI.',
  StaffRole.receptionist: 'Front desk — patients, appointments, invoices, and payments.',
  StaffRole.labStaff: 'Laboratory — view patients and upload visit attachments.',
};
